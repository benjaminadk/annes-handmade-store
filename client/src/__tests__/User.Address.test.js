import React from 'react'
import UserAddress from '../components/User.Address'
import { shallow } from 'enzyme'

const ships = [
  {
    city: 'Lake George',
    email: 'test@test.com',
    firstName: 'Ben',
    id: '1',
    lastName: 'Brooke',
    state: 'California',
    street1: '123 Fake Lane',
    street2: '',
    title: 'My Shipping',
    zip: '12845'
  }
]

const bills = [
  {
    city: 'Lake George',
    email: 'test@test.com',
    firstName: 'Ben',
    id: '1',
    lastName: 'Brooke',
    state: 'California',
    street1: '123 Fake Lane',
    street2: '',
    title: 'My Shipping',
    zip: '12845'
  }
]

describe('User Address Component', () => {
  const wrapper = shallow(<UserAddress ships={ships} bills={bills} />)
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('accects a ships prop', () => {
    expect(wrapper.props().ships).toEqual(ships)
  })
  it('accecpts a bills prop', () => {
    expect(wrapper.props().bills).toEqual(bills)
  })
})
