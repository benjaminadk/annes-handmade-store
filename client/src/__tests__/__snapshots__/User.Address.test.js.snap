// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Address Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(UserAddress)
    bills={
        Array [
            Object {
              "city": "Lake George",
              "email": "test@test.com",
              "firstName": "Ben",
              "id": "1",
              "lastName": "Brooke",
              "state": "California",
              "street1": "123 Fake Lane",
              "street2": "",
              "title": "My Shipping",
              "zip": "12845",
            },
          ]
    }
    ships={
        Array [
            Object {
              "city": "Lake George",
              "email": "test@test.com",
              "firstName": "Ben",
              "id": "1",
              "lastName": "Brooke",
              "state": "California",
              "street1": "123 Fake Lane",
              "street2": "",
              "title": "My Shipping",
              "zip": "12845",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "bills": Array [
        Object {
          "city": "Lake George",
          "email": "test@test.com",
          "firstName": "Ben",
          "id": "1",
          "lastName": "Brooke",
          "state": "California",
          "street1": "123 Fake Lane",
          "street2": "",
          "title": "My Shipping",
          "zip": "12845",
        },
      ],
      "classes": Object {
        "addresses": "UserAddress-addresses-3",
        "list": "UserAddress-list-4",
        "root": "UserAddress-root-1",
        "title": "UserAddress-title-2",
      },
      "ships": Array [
        Object {
          "city": "Lake George",
          "email": "test@test.com",
          "firstName": "Ben",
          "id": "1",
          "lastName": "Brooke",
          "state": "California",
          "street1": "123 Fake Lane",
          "street2": "",
          "title": "My Shipping",
          "zip": "12845",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "bills": Array [
          Object {
            "city": "Lake George",
            "email": "test@test.com",
            "firstName": "Ben",
            "id": "1",
            "lastName": "Brooke",
            "state": "California",
            "street1": "123 Fake Lane",
            "street2": "",
            "title": "My Shipping",
            "zip": "12845",
          },
        ],
        "classes": Object {
          "addresses": "UserAddress-addresses-3",
          "list": "UserAddress-list-4",
          "root": "UserAddress-root-1",
          "title": "UserAddress-title-2",
        },
        "ships": Array [
          Object {
            "city": "Lake George",
            "email": "test@test.com",
            "firstName": "Ben",
            "id": "1",
            "lastName": "Brooke",
            "state": "California",
            "street1": "123 Fake Lane",
            "street2": "",
            "title": "My Shipping",
            "zip": "12845",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
